crates/hyperswitch_connectors/src/connectors/taxjar/transformers.rs

```diff
@@ -49,32 +61,64 @@ pub struct TaxjarCard {
     complete: bool,
 }
 
-impl TryFrom<&TaxjarRouterData<&PaymentsAuthorizeRouterData>> for TaxjarPaymentsRequest {
+impl TryFrom<&TaxjarRouterData<&types::PaymentsTaxCalculationRouterData>>
+    for TaxjarPaymentsRequest
+{
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
-        item: &TaxjarRouterData<&PaymentsAuthorizeRouterData>,
+        item: &TaxjarRouterData<&types::PaymentsTaxCalculationRouterData>,
     ) -> Result<Self, Self::Error> {
-        match item.router_data.request.payment_method_data.clone() {
-            PaymentMethodData::Card(req_card) => {
-                let card = TaxjarCard {
-                    number: req_card.card_number,
-                    expiry_month: req_card.card_exp_month,
-                    expiry_year: req_card.card_exp_year,
-                    cvc: req_card.card_cvc,
-                    complete: item.router_data.request.is_auto_capture()?,
-                };
+        let request = &item.router_data.request;
+        let currency = item.router_data.request.currency;
+        let currency_unit = &api::CurrencyUnit::Base;
+        let shipping = &item
+            .router_data
+            .request
+            .shipping_address
+            .address
+            .clone()
+            .ok_or(errors::ConnectorError::MissingRequiredField {
+                field_name: "address",
+            })?;
```

Comment:
How are missing_required_params propagated in the api response?

We should give 2xx or SDK should handle a 4xx Scenario? @jarnura

Reply:
In my opinion, The api is supposed to calculate the tax with the given inputs. If it is not able to calculate it for some reason, it should not send a 2xx response. Missing required fields can occur because of 2 cases
- Integration errors
We expected some field to be sent from the merchant in the payments create request or while creating the business profile, these should be caught in the integration phase.

- Insufficient data from the customer / wallet 
Some fields that are required by the underlying connector to calculate the tax are not received by the api. You check this case by testing the flow with different wallet providers. We can throw 4xx if it can be handled by the sdk to collect these fields.